app/org/sagebionetworks/bridge/stormpath/StormpathDirectoryDao.java

```diff
@@ -177,9 +177,9 @@ private static AccountStoreMapping getApplicationMapping(String href, Applicatio
     private void adjustPasswordPolicies(Study study, Directory directory) {
         PasswordPolicy passwordPolicy = directory.getPasswordPolicy();
         
-        // According to the documentation, there is only one...
-        ModeledEmailTemplate template = passwordPolicy.getResetEmailTemplates().iterator().next();
-        updateTemplate(study, template, study.getResetPasswordTemplate(), "resetPassword");
+        for (ModeledEmailTemplate template : passwordPolicy.getResetEmailTemplates()) {
+            updateTemplate(study, template, study.getResetPasswordTemplate(), "resetPassword");
+        }
```

Comment:
After following up with Alx, we suspect that hasNext() has a side effect, so we want to do this in a loop rather than calling iterator.next().

That said, our unit tests assume there's only ever 1 template, and our implicit assumption is that there's only ever 1 template. We should make this assumption explicit, check for the number of templates, and log an error if there isn't.